blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
The one in XAResourceManager if for recovering branches and the one in HttpTxManager is for active branches. I think I separated them out because XAResourceManager is the class that drives recovery (and not HttpTxManager). If they need to be in synch then maybe we could revisit the design (but I am not sure they do need to be in synch).

You said TestTransactions::test_recovery() failed with SIGABORT - do you know which bit of code was responsible for that?